engine/app/lib/citizens_advice_components/elements/base.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require "active_support/core_ext/module/delegation"
+
+module CitizensAdviceComponents
+  module Elements
+    class Base
+      include ActionView::Context
+
+      delegate :content_tag, :tag, :safe_join, :link_to, :capture, to: :@template
+
+      attr_reader :template, :object, :attribute
+      attr_accessor :options
+
+      def initialize(template, object, attribute, **kwargs)
+        @template = template
+        @object = object
+        @attribute = attribute
+
+        @options = kwargs.with_defaults(default_options)
+      end
+
+      def render
+        # NO OP
+      end
+
+      private
+
+      def current_value
+        object.send(attribute)
+      end
+
+      def default_options
+        {}
+      end
+
+      def error_message
+        object.errors.full_messages_for(attribute)&.first
+      end
+
+      def object_name
+        object.to_model.model_name.singular
+      end
+
+      def field_name
+        # We still support Rails 6.1 but rely on the use of field_name
+        # so we provide a shim for this method for older versions of Rails.
+        if Rails::VERSION::MAJOR < 7
+          field_name_shim(object_name, attribute)
+        else
+          template.field_name(object_name, attribute)
+        end
+      end
+
+      def field_id
+        # We still support Rails 6.1 but rely on the use of field_id
+        # so we provide a shim for this method for older versions of Rails.
+        if Rails::VERSION::MAJOR < 7
```

Comment:
Given the engine's gemspec requires Rails > 7.1, should we remove this? Same with `field_name`.